src/main/java/org/jenkinsci/plugins/ewm/actions/ExwsAllocateActionImpl.java

```diff
@@ -83,9 +84,13 @@ public ExternalWorkspace getDynamic(String token) {
     @CheckForNull
     private ExternalWorkspace getExternalWorkspaceById(String id) {
         if (workspacesMap == null || workspacesMap.size() != allocatedWorkspaces.size()) {
```

Comment:
🐛 workspacesMap.size() is not thread-safe as well. I would just sync the entire method